service/controller/resource/prometheusremotewrite/delete.go

```diff
@@ -31,27 +33,58 @@ func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
 		}
 
 		for _, current := range prometheusList.Items {
```

Comment:
This is bad because the loop won't continue and other Prometheus would not be unconfigured. But without this, errors are ignored which is also bad.
We need a way to catch this error when we remove this check